ui/src/data-services/hooks/deployments/utils.ts

```diff
@@ -12,7 +12,9 @@ export const convertToFormData = (fieldValues: DeploymentFieldValues) => {
     longitude: fieldValues.longitude,
     research_site_id: fieldValues.siteId,
   }).forEach(([key, value]) => {
-    data.append(key, `${value}`)
+    if (value && value !== 'undefined') {
```

Comment:
@annavik I am curious how these variables are getting converted to the string "undefined". Any ideas?

Reply:
Actually, the problem is the line below. JavaScript behaves like this:
<img width="143" alt="Screenshot 2024-03-18 at 22 38 53" src="https://github.com/RolnickLab/ami-platform/assets/11680517/49b4808d-5768-4e0b-9764-87778dc8275f">

You can try it out in the console!

Reply:
If we want to make sure we will not pass the string "undefined" to server, we must make sure the value is not undefined before we try make a string out of it. Pretty much what you did, but the string comparison is not necessary. I would also make sure we will not ignore numeric values equal to 0 by making the check a bit more specific.

```suggestion
    if (value !== undefined) {
```

Reply:
You know because
<img width="123" alt="Screenshot 2024-03-18 at 22 44 50" src="https://github.com/RolnickLab/ami-platform/assets/11680517/6d0f97f1-2f55-4dc4-b210-bab84423e4dd">

Reply:
> If we want to make sure we will not pass the string "undefined" to server, we must make sure the value is not undefined before we try make a string out of it. Pretty much what you did, but the string comparison is not necessary. I would also make sure we will not ignore numeric values equal to 0 by making the check a bit more specific.

This is what I started with, but somehow the undefined value is converted to a string before this function is called. So `value !== undefined` does not catch it :(